deeplake/core/chunk_engine.py

```diff
@@ -714,63 +722,149 @@ def _samples_to_chunks(
             )  # type: ignore
             if register_creds:
                 self.register_new_creds(num_samples_added, samples)
             if num_samples_added == 0:
-                current_chunk = self._create_new_chunk(register, row=start_chunk_row)
+                current_chunk = self._create_new_chunk(
+                    register and start_chunk_row is not None, row=start_chunk_row
+                )
+                current_chunk._update_tensor_meta_length = False
                 if start_chunk_row is not None:
                     start_chunk_row += 1
+                elif register:
+                    enc_ids.append(current_chunk.id)
+                    enc_count.append(0)
                 updated_chunks.append(current_chunk)
             elif num_samples_added == PARTIAL_NUM_SAMPLES:
                 sample = samples[0]
-                if register and sample.is_first_write:
-                    enc.register_samples(1)
-                if sample.is_last_write:
+                if sample.is_first_write:
                     if register:
-                        self.tile_encoder.register_sample(sample, self.num_samples - 1)
-                        if update_commit_diff:
-                            commit_diff.add_data(1)
-                    else:
-                        tiles[nsamples - len(samples)] = (
-                            sample.sample_shape,
-                            sample.tile_shape,
-                        )
+                        if start_chunk_row is not None:
+                            enc.register_samples(1)
+                        else:
+                            enc_count[-1] += 1
+                if sample.is_last_write:
+                    tiles[
+                        incoming_num_samples
+                        - len(samples)
+                        + bool(register) * orig_meta_length
+                    ] = (
+                        sample.sample_shape,
+                        sample.tile_shape,
+                    )
                     samples = samples[1:]
+                    if lengths is not None:
+                        lengths = lengths[1:]
+                    num_samples_added = 1
+                    num_samples += 1
+                else:
+                    num_samples_added = 0
                 if len(samples) > 0:
                     current_chunk = self._create_new_chunk(
-                        register, row=start_chunk_row
+                        register and start_chunk_row is not None, row=start_chunk_row
                     )
+                    current_chunk._update_tensor_meta_length = False
                     if start_chunk_row is not None:
                         start_chunk_row += 1
+                    elif register:
+                        enc_ids.append(current_chunk.id)
+                        enc_count.append(0)
                     updated_chunks.append(current_chunk)
+            elif num_samples_added == FAST_EXTEND_BAIL:
```

Comment:
This happens when numpy optimization fails. We convert incoming samples to list and try again.